forui/lib/src/widgets/pagination/pagination_controller.dart

```diff
@@ -0,0 +1,192 @@
+    final value = hasClients && page != null ? page!.round() : _value;
+    if (0 < value) {
+      _value = value - 1;
+
+      if (hasClients) {
+        super.previousPage(duration: const Duration(milliseconds: 300), curve: Curves.ease);
+      }
+      notifyListeners();
+    }
+  }
+
+  /// Moves to the next page if the current page is less than the total number of pages.
+  void next() {
+    final value = hasClients && page != null ? page!.round() : _value;
+    if (value < length - 1) {
+      _value = value + 1;
+
+      if (hasClients) {
+        super.nextPage(duration: const Duration(milliseconds: 300), curve: Curves.ease);
+      }
+      notifyListeners();
+    }
+  }
+
+  /// Returns the current page index.
+  int get value => _value;
+
+  /// Sets the current page index.
+  set value(int index) {
+    if (0 <= index && index <= length) {
+      _value = index;
+
+      if (hasClients) {
+        super.jumpToPage(index);
+      }
+      notifyListeners();
+    } else {
+      throw StateError('The index must be within the allowed range.');
+    }
+  }
+
+  @override
+  void dispose() {
+    removeListener(_handleScrollChanges);
+    super.dispose();
+  }
+
+  /// Calculates the range of page numbers to display around the current page.
+  ///
+  /// Returns a tuple of the start and end page numbers by centering the range around the current page,
+  /// ensuring a balanced display of sibling pages on either side while considering pagination constraints.
+  (int, int) calculateSiblingRange() {
+    final int rangeStart;
+    final int rangeEnd;
+    if (length.isFinite) {
+      final last = length.toInt() - 1;
+      if (length <= minPagesDisplayedAtEdges) {
+        return (0, last);
+      }
+      rangeStart =
+          _value - siblings < 0
+              ? 0
+              : _value > (last - minPagesDisplayedAtEdges)
+              ? (last - minPagesDisplayedAtEdges) - siblings
+              : _value <= minPagesDisplayedAtEdges
+              ? 0
+              : _value - siblings;
+
+      rangeEnd =
+          _value + siblings > last
+              ? last
+              : _value < minPagesDisplayedAtEdges + 1
+              ? minPagesDisplayedAtEdges + siblings
+              : _value >= (last - minPagesDisplayedAtEdges)
+              ? last
+              : _value + siblings;
+    } else {
+      rangeStart =
+          _value - siblings < 0
+              ? 0
+              : _value <= minPagesDisplayedAtEdges
+              ? 0
+              : _value - siblings;
+      rangeEnd = _value < minPagesDisplayedAtEdges + 1 ? minPagesDisplayedAtEdges : _value + siblings;
+    }
+    return (rangeStart, rangeEnd);
+  }
+
+  /// Returns the minimum number of pages to display at both the start and end of the pagination.
+  ///
+  /// If the total number of pages is too small to accommodate both the edge pages
+  /// and the full set of sibling pages around the current page, all pages are displayed instead.
+  @internal
```

Comment:
Can we get away with making this private?

Otherwise, if this _must be internal, we should move it into an internal extension
```dart
@internal
extension InternalExtension on FPaginationController {
  // internal methods
}
```